plugins/languages/treesitter/treesitter.nix

```diff
@@ -6,210 +6,169 @@
+        Whether to enable treesitter highlighting.
+      '';
+
+      disable = helpers.mkStrLuaFnOr (types.listOf types.str) ''
+        Can either be a list of the names of parsers you wish to disable or
+        a lua function that returns a boolean indicating the parser should be disabled.
+      '';
+
+      custom_captures = helpers.defaultNullOpts.mkAttrsOf types.str { } ''
+        Custom capture group highlighting.
+      '';
+    };
+
+    incremental_selection = {
+      enable = helpers.defaultNullOpts.mkBool true ''
+        Incremental selection based on the named nodes from the grammar.
+      '';
+
+      keymaps =
         let
-          keymap =
-            default:
-            mkOption {
-              type = types.str;
-              inherit default;
-              description = "Key shortcut";
-            };
+          keymap = default: helpers.defaultNullOpts.mkStr default ''Key shortcut'';
         in
         {
-          enable = mkEnableOption "incremental selection based on the named nodes from the grammar";
-          keymaps = {
-            initSelection = keymap "gnn";
-            nodeIncremental = keymap "grn";
-            scopeIncremental = keymap "grc";
-            nodeDecremental = keymap "grm";
-          };
+          init_selection = keymap "gnn";
+          node_incremental = keymap "grn";
+          scope_incremental = keymap "grc";
+          node_decremental = keymap "grm";
         };
+    };
 
-      indent = mkEnableOption "tree-sitter based indentation";
-
-      folding = mkEnableOption "tree-sitter based folding";
-
-      languageRegister = mkOption {
-        type = with types; attrsOf (either str (listOf str));
-        description = ''
-          This is a wrapping of the `vim.treesitter.language.register` function.
-          Register specific parsers to one or several filetypes.
-          The keys are the parser names and the values are either one or several filetypes.
-        '';
-        default = { };
-        example = {
-          cpp = "onelab";
-          python = [
-            "myFiletype"
-            "anotherFiletype"
-          ];
-        };
-      };
+    indent = {
+      enable = helpers.defaultNullOpts.mkBool true ''
+        Whether to enable tree-sitter based indentation.
+      '';
+    };
 
-      grammarPackages = mkOption {
-        type = with types; listOf package;
-        default = cfg.package.passthru.allGrammars;
-        description = "Grammar packages to install";
-      };
+    folding = {
+      enable = helpers.defaultNullOpts.mkBool true ''
+        Whether to enable tree-sitter based folding.
+      '';
+    };
 
-      moduleConfig = mkOption {
-        type = types.attrsOf types.anything;
-        default = { };
-        description = "This is the configuration for extra modules. It should not be used directly";
-      };
+    # TODO: default null nixgrammars
+    ensure_installed = helpers.defaultNullOpts.mkListOf types.str [ "all" ] ''
```

Comment:
Is this an upstream treesitter option or a custom nixvim option?

`settingsOptions` is only for stuff that should be passed to the plugins `setup` function; anything specific to nixvim should go in `extraOptions` instead.

Reply:
It's upstream